From a9eab9d87b6c5e0eb5142d4004727b78749ca5d0 Mon Sep 17 00:00:00 2001
From: Joe Slater <jslater@windriver.com>
Date: Sun, 8 Feb 2026 21:06:44 -0800
Subject: [PATCH] dpkg: fix configuration issue for mips64

configure cannot determine the proper cpu, os, or
architecture for mips64, and possibly other arch's
because of faulty code added to Arch.pm in the latest
release from upstream.  We remove that code.

Upstream-Status: Inappropriate [not a debian architecture]

Signed-off-by: Joe Slater <jslater@windriver.com>

Rebase to 1.23.5
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 scripts/Dpkg/Arch.pm | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/scripts/Dpkg/Arch.pm b/scripts/Dpkg/Arch.pm
index 3d9e3e97503b..05c6debc3cc7 100644
--- a/scripts/Dpkg/Arch.pm
+++ b/scripts/Dpkg/Arch.pm
@@ -326,9 +326,6 @@ sub _load_tupletable
                     (my $dt = $debtuple) =~ s/<cpu>/$_cpu/;
                     (my $da = $debarch) =~ s/<cpu>/$_cpu/;
 
-                    next if exists $debarch_to_debtuple{$da}
-                         or exists $debtuple_to_debarch{$dt};
-
                     $debarch_to_debtuple{$da} = $dt;
                     $debtuple_to_debarch{$dt} = $da;
                 }
